### PR TITLE
164 applies to type optional

### PR DESCRIPTION
**Pull request by njogz**
# Description

This PR makes a change to how `appliesToType` is processed such that if it is absent tthere is no effect and cards and fields should be displayed by default.

medic/medic-conf#164

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: src/contact-summary/contact-summary-emitter.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ function emitter(contactSummary, contact, reports) {
       throw new Error("You cannot set appliesToType to an array which includes the type 'report' and another type.");
     }
     
-    if (appliesToType.includes('report')) {
+    if (appliesToType.includes('report') || (appliesToType.length === 0 && typeof card.fields === 'function')) {
```
Comment (garethbowen) [inline src/contact-summary/contact-summary-emitter.js:35]: I think `"report"` is a special case. It doesn't make sense to have a appliesToType of "report" and "person" and line 32 will throw an error on this case.

The original issue was mostly concerned about matching any contact, so if you revert this line it will fall through to line 48 which you've modified to add the card assuming it's related to a contact.

File: src/contact-summary/contact-summary-emitter.js [hunk 2]
```diff
@@ -81,6 +84,9 @@ function execAppliesIf(prop, report) {
 }
 
 function addCard(card, context, r) {
+  if (typeof card.fields === 'function' && arguments.length < 3) {
+    return;
+  }
```
Comment (garethbowen) [inline src/contact-summary/contact-summary-emitter.js:89]: I'm not sure what you're trying to achieve here. Can `fields` only be a function when adding a card about a report? Can we revert this change?

**Review by garethbowen — CHANGES_REQUESTED** [review]
Sweet! I've put a couple of comments inline...

File: src/contact-summary/contact-summary-emitter.js (hunk 2 above)
Comment (njogz) [reply src/contact-summary/contact-summary-emitter.js:89]: Yes. This can be reverted. Yes, that was the impression I got from line 102 about fields and report cards.

**Review by garethbowen — APPROVED** [review]
Awesome! Nice work.
